warehouse/query-core/src/main/java/datawave/query/iterator/QueryOptions.java

```diff
@@ -2259,4 +2260,290 @@ public Equality getEquality() {
+     *            The iterator setting to add the option to.
+     * @param option
+     *            The name of the option to add.
+     * @param value
+     *            The value of the option to add.
+     * @param valueTransformer
+     *            A function that converts the value to a string.
+     * @param allowBlankValues
+     *            If true, blank values will be converted to a single space, as blank values will fail in InputFormatBase when run through the MapReduce api.
+     */
+    public static <T> void addOption(IteratorSetting setting, String option, T value, Function<T,String> valueTransformer, boolean allowBlankValues) {
+        // If we have a default options implementation for the specified iterator setting's class, fetch it.
+        DefaultOptions defaultOptions = getDefaultOptions(setting.getIteratorClass());
+        // If the value matches the default value, do not add it to the setting.
+        if (defaultOptions.hasDefaultValue(option) && defaultOptions.equalsDefaultValue(option, value)) {
+            setting.removeOption(option);
+            return;
+        }
+        // Otherwise convert it, and if blank and blank values allowed, change it to something else, or it will fail in InputFormatBase when run through the
+        // MapReduce api.
+        String valueString = valueTransformer.apply(value);
+        if (allowBlankValues && org.apache.commons.lang.StringUtils.isBlank(valueString)) {
+            valueString = " ";
+        }
+        setting.addOption(option, valueString);
+    }
+
+    /**
+     * Default options for QueryOptions. TODO: Make immutable with a builder that is used in createDefaultOptions().
+     */
+    protected static final class DefaultOptions {
+
+        public final Map<String,Object> defaultValues;
+
+        private DefaultOptions() {
+            this.defaultValues = MapUtils.unmodifiableMap(new HashMap<>());
+        }
+
+        public DefaultOptions(Map<String,Object> defaultValues) {
+            this.defaultValues = MapUtils.unmodifiableMap(defaultValues);
+        }
+
+        public boolean hasDefaultValue(String option) {
+            return defaultValues.containsKey(option);
+        }
+
+        public boolean equalsDefaultValue(String option, Object value) {
+            if (hasDefaultValue(option)) {
+                return Objects.equals(defaultValues.get(option), value);
+            }
+            return false;
+        }
+
+        public static Builder builder() {
+            return new Builder();
+        }
+
+        protected static class Builder {
+            private final Map<String,Object> values;
+
+            public Builder() {
+                values = new HashMap<>();
+            }
+
+            public Builder putDefaultValue(String option, Object value) {
+                if (value.getClass().isAssignableFrom(Collection.class)) {
+                    value = Collections.unmodifiableCollection((Collection<?>) value);
```

Comment:
Upon further thinking, this may cause issues with equalsDefaultValue() when trying to compare the stored immutable collections and any passed-in lists/sets. Can you add a test case to see equalsDefaultValues returns true when a set or list is passed in with default values?